File: api-tester/src/main/java/com/revenuecat/apitester/java/PeriodAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.Period;

@SuppressWarnings({"unused", "SpellCheckingInspection"})
final class PeriodAPI {
    static void check(final Period period) {
        Period newPeriod = Period.Factory.create("P1M");
        int val = period.getValue();
        Period.Unit unit = period.getUnit();
        String iso8601 = period.getIso8601();
        Double valueInMonths = period.getValueInMonths();

        switch (unit) {
            case DAY:
            case WEEK:
            case MONTH:
            case YEAR:
            case UNKNOWN:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PeriodAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.Period

@Suppress("unused", "UNUSED_VARIABLE")
private class PeriodAPI {
    fun check(period: Period) {
        with(period) {
            val value: Int = value
            val unit: Period.Unit = unit
            val iso8601: String = iso8601
            val valueInMonths: Double = valueInMonths

            when (unit) {
                Period.Unit.DAY,
                Period.Unit.WEEK,
                Period.Unit.MONTH,
                Period.Unit.YEAR,
                Period.Unit.UNKNOWN,
                -> {
                }
            }.exhaustive
        }
    }
}
